internal/kafka/internal/services/kafka.go

```diff
@@ -1150,6 +1214,73 @@ func buildManagedKafkaCR(kafkaRequest *dbapi.KafkaRequest, kafkaConfig *config.K
 	return managedKafkaCR, nil
 }
 
+// buildReservedManagedKafkaCR buklds a Reserved Managed Kafka CR.
+// The ID, K8s object ID, K8s namespace and PlacementID are all set to
+// the provided kafkaID.
+func (k *kafkaService) buildReservedManagedKafkaCR(kafkaID string, instanceType string, desiredStrimziVersion api.StrimziVersion, kafkaConfig *config.KafkaConfig) (*managedkafka.ManagedKafka, *errors.ServiceError) {
+	// Reserved instances always make use of the streaming base unit which is x1.
+	// For now we hardcode it but there might be a better alternative to it.
+	streamingBaseUnit := "x1"
+	kafkaInstanceSize, err := kafkaConfig.GetKafkaInstanceSize(instanceType, streamingBaseUnit)
+	if err != nil {
+		return nil, errors.NewWithCause(errors.ErrorGeneral, err, "unable to list kafka request")
+	}
+	labels := map[string]string{
+		"bf2.org/kafkaInstanceProfileQuotaConsumed": strconv.Itoa(kafkaInstanceSize.QuotaConsumed),
+		"bf2.org/kafkaInstanceProfileType":          instanceType,
+		"bf2.org/deployment":                        "reserved",
+	}
+
+	desiredKafkaVersion := desiredStrimziVersion.GetLatestKafkaVersion()
+	if desiredKafkaVersion == nil {
+		return nil, errors.GeneralError("No available Kafka versions in Strimzi version %s", desiredStrimziVersion.Version)
+	}
+	desiredKafkaIBPVersion := desiredStrimziVersion.GetLatestKafkaIBPVersion()
+	if desiredKafkaIBPVersion == nil {
+		return nil, errors.GeneralError("No available Kafka IBP versions in Strimzi version %s", desiredStrimziVersion.Version)
+	}
+
+	managedKafkaCR := &managedkafka.ManagedKafka{
+		Id: kafkaID,
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "ManagedKafka",
+			APIVersion: "managedkafka.bf2.org/v1alpha1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      kafkaID,
+			Namespace: kafkaID,
+			Annotations: map[string]string{
+				"bf2.org/id":          kafkaID,
+				"bf2.org/placementId": kafkaID,
+			},
+			Labels: labels,
+		},
+		Spec: managedkafka.ManagedKafkaSpec{
+			Capacity: managedkafka.Capacity{
+				IngressPerSec:               kafkaInstanceSize.IngressThroughputPerSec.String(),
+				EgressPerSec:                kafkaInstanceSize.EgressThroughputPerSec.String(),
+				TotalMaxConnections:         kafkaInstanceSize.TotalMaxConnections,
+				MaxDataRetentionSize:        kafkaInstanceSize.MaxDataRetentionSize.String(),
+				MaxPartitions:               kafkaInstanceSize.MaxPartitions,
+				MaxDataRetentionPeriod:      kafkaInstanceSize.MaxDataRetentionPeriod,
+				MaxConnectionAttemptsPerSec: kafkaInstanceSize.MaxConnectionAttemptsPerSec,
+			},
+			Endpoint: managedkafka.EndpointSpec{
+				BootstrapServerHost: fmt.Sprintf("%s-dummyhost", kafkaID),
+			},
+			Versions: managedkafka.VersionsSpec{
+				Kafka:    desiredKafkaVersion.Version,
+				Strimzi:  desiredStrimziVersion.Version,
+				KafkaIBP: desiredKafkaIBPVersion.Version,
+			},
+			Deleted: false,
+			Owners:  []string{}, // TODO is this enough?
```

Comment:
From what I see, this is enough. 
The ManagedKafkaCR is created and there are no errors.